app/models/shortened_link.rb

```diff
@@ -0,0 +1,4 @@
+class ShortenedLink < ApplicationRecord
+  validates :name, :shortened_path, :original_url, presence: true
+  validates :shortened_path, uniqueness: true
```

Comment:
Wouldn't this `shortened_path` attribute potentially clash with a `shortened_path` route defined in the routes file? Same for `original_url`

Reply:
Could be a case, but I don't think anybody would define a route called `/shortened` or `/original`. Still open to change for better names though I think these are clear for the reader